lib/pure/strutils.nim

```diff
@@ -154,6 +154,124 @@ proc isDigit*(c: char): bool {.noSideEffect,
     doAssert isDigit('8') == true
   return c in Digits
 
+func isNumeric*(s: string, enableNaNInf, enableLooseDot = false): bool =
```

Comment:
talking about efficiency, `s/string/openArray[char]/`
(avoids copies in some cases)